src/main/java/org/broadinstitute/hellbender/tools/copynumber/arguments/CopyNumberArgumentValidationUtils.java

```diff
@@ -48,17 +48,24 @@ public static void validateIntervalArgumentCollection(final IntervalArgumentColl
     }
 
     /**
-     * Validate that a list of locatables is valid and sorted according to a sequence dictionary and contains no duplicates or overlaps.
+     * Validate that a list of locatables is valid and sorted according to a sequence dictionary
```

Comment:
Note that use of non-overlapping intervals for all CNV locatable collections was enforced by using the base AbstractLocatableCollection class---see documentation of that class and subclasses. In practice, PreprocessIntervals creates valid bins, which remain valid as they move through the pipelines.

It might be kind of a nightmare to go through and change all the relevant documentation---and even more of a nightmare to define behavior for algorithms/procedures, such as segmentation or segment merging, which rely on this property (part of why we enforce this in the first place).

I haven't looked closely at the code to see why you absolutely need to use the CNV collections classes to represent overlapping intervals, but I'd strongly suggest refactoring to avoid this if possible. Perhaps instead convert to VCF, IntervalList, etc.?

Reply:
Took a quick look---you only need to validate the sample name (which might be a redundant check, haven't looked closely) and read in a list of Locatables from copyNumberSegmentsFile. So unless you need the other machinery that the CNV collection class buys you, you might get away with doing something more one-off for this procedure---perhaps a private method that uses one of the other more generic TSV reading utility methods.

In any case, I'd add some comments/documentation that indicates that this business with overlapping intervals is the exception, not the norm.

Reply:
I was hoping to be able to use the encoder and decoder to get my data into a format compatible with FuncotateSegments without writing more IO methods, but that didn't work out so well because FuncotateSegments expects fields that the Python segment generation code doesn't write out.  I'm still struggling with an elegant way to turn the VCF into something that can be annotated, so I may end up ripping out the CopyNumberSegmentCollection-related changes anyway.

Reply:
I thought FuncotateSegments was only intended to be used on somatic seg files, not the VCFs produced by the gCNV pipeline? Is there any additional functionality in FuncotateSegments besides being able to read seg files---can you not use Funcotator instead?

I was always in favor of having the somatic pipeline output VCFs rather than seg files (there are probably still a few related issues filed), but not sure what the plan going forward should be.

Reply:
Was reminded of this while looking at #6924. Not sure if you were planning on going back and addressing it, but I would make sure that all concomitant changes are reverted before merging. Happy to help review or discuss more!

Reply:
Sorry, just to be clear, I meant the changes to AbstractLocatableCollection, etc. to allow overlapping intervals should be rolled back, since most of the existing methods and code very strongly depend on this assumption. I think it will be difficult to update code and documentation if you want to keep those changes. I don’t think I’m up to date on the development plans for Funcotator/FuncotateSegments, so I don’t have strong opinions there.